cuda.Dockerfile

```diff
@@ -0,0 +1,61 @@
+# syntax=docker/dockerfile:1.5
+# Mounted gpu has cuda v11.5
+FROM nvcr.io/nvidia/cuquantum-appliance:22.03-cirq
+ARG SDK_VERSION
+
+WORKDIR /app
+
+# https://askubuntu.com/questions/1408016/the-following-signatures-couldnt-be-verified-because-the-public-key-is-not-avai
+RUN <<EOF
+apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/3bf863cc.pub
+apt update --yes
+apt install --yes wget build-essential gcc git
+apt install --yes python3-pip
+EOF
+
+#download cuquantum from https://developer.nvidia.com/cuquantum-downloads?target_os=Linux&target_arch=x86_64&Distribution=Ubuntu&target_version=20.04&target_type=deb_local
+RUN <<EOF
+wget https://developer.download.nvidia.com/compute/cuquantum/22.07.0/local_installers/cuquantum-local-repo-ubuntu2004-22.07.0_1.0-1_amd64.deb
+dpkg -i cuquantum-local-repo-ubuntu2004-22.07.0_1.0-1_amd64.deb
+cp /var/cuquantum-local-repo-ubuntu2004-22.07.0/cuquantum-*-keyring.gpg /usr/share/keyrings/
+apt update --yes
+apt install --yes cuquantum cuquantum-dev cuquantum-doc
+EOF
+
+RUN <<EOF
+export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/opt/nvidia/cuquantum/lib/
+echo "export PATH=/usr/local/cuda-11.5/bin${PATH:+:${PATH} }" >> ~/.bashrc
+. ~/.bashrc
+EOF
+
+ENV CUQUANTUM_DIR=/opt/nvidia/cuquantum
+
+# get required tools to build qsim
+RUN <<EOF
+python -m pip install pybind11
+apt install --yes git
+export DEBIAN_FRONTEND=noninteractive
+apt install --yes cmake
+EOF
+
+# install qsimcirq and orquestra SDK
+RUN <<EOF
+git clone https://github.com/quantumlib/qsim.git
+cd qsim
+make clean
+make
+python -m pip install -e .
```

Comment:
I was talking with @brian-dellabetta about this, I wonder if we really need to do this from scratch? It adds a significant amount of time to the build process for the notebooks. If the pre-built wheels don't exist, or aren't adequate, my proposal is to build it once, and then upload our "custom" wheel to our private Nexus PyPI repository.

Reply:
It sounds like this could be improved later on. Hence I suggest to not do it here (to avoid scope creep) and improve it after this PR gets merged.

Reply:
Another alternative to consider in the future is split the `qsim` and `orquestra-sdk` installations to separate Docker layers + pin `qsim`'s version so that its installation would be cached.